src/main/blackbox/blackbox_io.c

```diff
@@ -470,7 +470,7 @@ bool blackboxDeviceOpen(void)
      *
      * 9 / 1250 = 7200 / 1000000
      */
-    blackboxWriteChunkSize = MAX((masterConfig.looptime * 9) / 1250, 4);
+    blackboxWriteChunkSize = MAX((GUARD_LOOPTIME * 9) / 1250, 4);
```

Comment:
(I was partially wrong in previous comment)
Attitude estimation just rotates 'DOWN' vector by gyro reading and very slowly mixes in actual ACC value. The rotation part is now quite costly - it uses full rotation matrix. Improving SIN/COS helps, but in fact input range very small (angle between two gyro readouts). 
Multiwii used small angle approximations, which was IMO very reasonable. Or second order may be implemented for this case.
Also using quaternion representation may help (and avoid using separate north/down vector)
Mixing with actual ACC is quite cheap, so not much gain is possible here ... 
Maybe gyro rate integration may be improved so larger time step is possible ... This is worth testing ...